src/Shared/CustomPagination/CustomPagination.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import { KeyboardEvent } from 'react';
+interface CustomPaginationProps {
+    totalPages: number;
+    currentPage: number;
+    onPageChange: (page: number) => void;
+}
+
+
+const CustomPagination: React.FC<CustomPaginationProps> = ({ totalPages, currentPage, onPageChange }) => {
+    const visiblePageLimit = 5; // Number of visible pages
+    const halfVisiblePageLimit = Math.floor(visiblePageLimit / 2);
+
+    const startPage = Math.max(1, currentPage - halfVisiblePageLimit);
+    const endPage = Math.min(totalPages, startPage + visiblePageLimit - 1);
+
+    const pages = Array.from({ length: endPage - startPage + 1 }, (_, index) => startPage + index);
+
+    const [pageNumberInput, setPageNumberInput] = useState('');
+
+    const handlePageNumberInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setPageNumberInput(e.target.value);
+    };
+
+    const handleGoToPage = () => {
+        const pageNumber = parseInt(pageNumberInput);
+        if (pageNumber >= 1 && pageNumber <= totalPages) {
+            onPageChange(pageNumber);
+        }
+
+    };
+    const handlePageNumberInputKeyPress = (e: KeyboardEvent<HTMLInputElement>) => {
+        if (e.key === 'Enter') {
+            handleGoToPage();
+        }
+    };
+
+    return (
+        <nav aria-label="..." className=''>
+            <ul className="pagination justify-content-center pagination-sm">
+                {currentPage > 1 && (
+                    <li className="page-item" onClick={() => onPageChange(currentPage - 1)}>
```

Comment:
The same goes here we should use either a button if we want use onClick, and we can also use the Link element if we will implement the pagination in the URL and fetch the data using the useParams hook from react-router-dom